src/auth/auth-config.ts

```diff
@@ -799,46 +799,51 @@ export class OIDCConfig implements OIDCAuthProviderConfig {
         '"OIDCAuthProviderConfig.clientSecret" must be a valid string.',
       );
     }
-    if (typeof options.responseType !== 'undefined') {
-      let idTokenType = false;
-      let codeType = false;
+    if (validator.isNonNullObject(options.responseType) && typeof options.responseType !== 'undefined') {
+      let idTokenType;
```

Comment:
Instead of a giant loop with many conditionals, how about this?

```typescript
Object.keys(options.responseType).forEach((key) => {
  if (!validResponseTypes.includes(key)) {
    // throw invalid parameter
  }
});

const idToken = options.responseType.idToken;
if (typeof idToken !== 'undefined') {
  if (!validator.isBoolean(idToken)) {
    // throw idToken must be a boolean
  }
}

const code = options.responseType.code;
if (typeof code !== 'undefined') {
  if (!validator.isBoolean(code)) {
    // throw code must be a boolean
  }
}

const allKeys = Object.keys(options.responseType).length;
const enabledCount = Object.values(options.responseType).filter(Boolean).length;
if (allKeys > 0 && enabledCount != 1) {
   // throw exactly one must be set
}
```

This allows following:
```
{}, {idToken: true}, {code: true}, {idToken: true, code: false}, {idToken: false, code: true}
```

Doesn't allow:
```
{idToken: false}, {code: false}, {idToken: true, code: true}, {idToken: false, code: false}
```

Reply:
I think {idToken: false}, {code: false} could be allowed if the request doesn't change the original responseType. In this API, since we're only verifying the correctness of options.responseType itself, we can't see whether we allow these without knowing about original OIDC config responseType.

Reply:
That makes sense. I'd suggest relaxing the check to allow those 2 cases:

```
if (allKeys > 1 && enabledCount != 1) {
   // throw exactly one must be set
}
```

Reply:
sounds good